### PR TITLE
HHH-11808 - Update 5.1 migration guide

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-11808

### CODE REVIEW

File: migration-guide.adoc [hunk 1]
```diff
@@ -32,3 +32,11 @@ These changes will only be a migration concern for applications directly using a
 * `org.hibernate.tool.hbm2ddl.SchemaValidator`
 * `org.hibernate.tool.schema.spi.SchemaManagementTool` or any of its delegates
 
+
```
Comment (gbadner) [inline migration-guide.adoc]: The following doesn't provide context. How about:

In 5.1.4, a new strategy for retrieving database tables was introduced that improves `SchemaMigrator` and `SchemaValidator` performance. This strategy executes a single `java.sql.DatabaseMetaData#getTables(String, String, String, String[])}` call to determine if each `javax.persistence.Entity` has a mapped database table. This strategy is the default, and uses the property setting `hibernate.hbm2ddl.jdbc_metadata_extraction_strategy=grouped`.  This strategy may require the `default_schema` and/or `default_catalog` to be provided.

To use the old strategy, which executes a `java.sql.DatabaseMetaData#getTables(String, String, String, String[])` call for each javax.persistence.Entity, use the property setting `hibernate.hbm2ddl.jdbc_metadata_extraction_strategy=individually`.

**Comment by gbadner** [comment]
The user guide should also be updated with this information.

The migration guide for 5.2 in master should also be updated, since the behavior changed starting in 5.2.3.

File: migration-guide.adoc (hunk 1 above)
Comment (dreab8) [inline migration-guide.adoc]: Hi @gbadner,
your comment looks perfect.
Thanks

**Comment by dreab8** [comment]
I'll update the user guide and 5.2 migration guide.

**Comment by gbadner** [comment]
Squashed and merged.

**Pull request closed without merging**